src/gwt/src/org/rstudio/studio/client/workbench/prefs/views/GeneralPreferencesPane.java

```diff
@@ -492,8 +503,24 @@ public boolean onApply(RPrefs rPrefs)
                                              restoreLastProject_.getValue());
          rPrefs.setProjectsPrefs(projectsPrefs);
       }
+      
+      if (restartRequired)
+      {
+         globalDisplay_.showYesNoMessage(
+               GlobalDisplay.MSG_QUESTION,
+               "Restart Required",
+               "You need to restart RStudio in order for these changes to take effect. " +
+               "Do you want to do this now?",
+               () -> {
+                  forceClosed(() -> {
+                     events_.fireEvent(new SwitchToProjectEvent(session_.getSessionInfo().getActiveProjectFile()));
```

Comment:
Does this actually cause the `rstudio.exe` to quit and restart, or just `rsession.exe`? (I think these changes require the former.) 

Does it work without a project?